### PR TITLE
Add Information Page

### PR DESCRIPTION
**Pull request by sanjeevz3009**
### What is the context of this PR?

JIRA Ticket: https://jira.ons.gov.uk/browse/CMS-179

Create an Information Page model in Wagtail CMS to allow content editors to manage structured information pages.

**Changes and Additions**
- HTML template created for the information page.
- Django Model created for Information Page
 
### How to review

- Spin up the project in your local machine
- Run the linters, code formatters and run the tests locally to make sure it's passing

**Create an information page**
- Go to `localhost:8000` 
- Navigate to the Wagtail admin site ` localhost:8000/admin`
- Within Wagtail Admin -> Pages -> Home -> Add Child Page
- Then you should be able to fill in the content of the information page and publish.

Untitled design

Screenshot 2024-11-07 at 17 30 10

Screenshot 2024-11-07 at 17 30 20

**Information Page Example**

Screenshot 2024-11-07 at 18 31 48

### Follow-up Actions

List any follow-up actions (if applicable), like needed documentation updates or additional testing.

### CODE REVIEW

File: cms/standard_pages/models.py
Comment (zerolab) [inline cms/standard_pages/models.py:33]:
```suggestion
    ]
    
    search_fields: ClassVar[list[index.SearchField | index.AutocompleteField]] = [
        *BasePage.search_fields,
        index.SearchField("summary"),
        index.SearchField("content")
    ]
```

with a `from wagtail.search import index` import at the top.

File: cms/standard_pages/models.py
Comment (zerolab) [inline cms/standard_pages/models.py:32]:
```suggestion
        FieldPanel("content"),
```

File: cms/standard_pages/models.py
Comment (zerolab) [inline cms/standard_pages/models.py:26]: nit: 
```suggestion
    content = StreamField(CoreStoryBlock())
```
so as to align with the other pages

File: cms/jinja2/templates/pages/information_page.html
Comment (zerolab) [inline cms/jinja2/templates/pages/information_page.html:9]:
```suggestion
    {% if page.summary %}
        <p>{{ page.summary }}</p>
    {% endif %}
```

File: cms/jinja2/templates/pages/information_page.html
Comment (zerolab) [inline cms/jinja2/templates/pages/information_page.html:5]: note: the template is missing the `last_updated` output

**Review by zerolab — CHANGES_REQUESTED** [review]
Great start!

Added a few notes and suggestions as a first pass.

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:14]: Since it's a small Class, I understand. I just summarised it now, hope that's okay.

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:32]: Updated!

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:26]: Updated!

File: cms/jinja2/templates/pages/information_page.html
Comment (sanjeevz3009) [reply cms/jinja2/templates/pages/information_page.html:9]: Corrected!

File: cms/jinja2/templates/pages/information_page.html
Comment (sanjeevz3009) [reply cms/jinja2/templates/pages/information_page.html:5]: Added.

File: cms/jinja2/templates/pages/information_page.html
Comment (MebinAbraham) [inline cms/jinja2/templates/pages/information_page.html:9]: This should be wrapped in a conditional, as the last updated field is optional. Currently, this would output `None`.

File: cms/standard_pages/migrations/0002_rename_body_informationpage_content.py
Comment (sanjeevz3009) [reply cms/standard_pages/migrations/0002_rename_body_informationpage_content.py:1]: I had everything in one migration file so far. It can be in one migration file that's no poblem, I left it for teaching purposes for Neha. Was going to make into one today before merging. Will update this once I go through everything with Neha =)

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:15]: Okay, makes sense! Habit from data science work where we had extensive doc strings but of course that was a difference scenario. Will fix that habit now!

File: cms/standard_pages/migrations/0002_rename_body_informationpage_content.py
Comment (sanjeevz3009) [reply cms/standard_pages/migrations/0002_rename_body_informationpage_content.py:1]: Will make sure to push 1 migration file in a PR as I could of made into 1 migration file and pushed it up. Then locally for teaching purposes I could of done something else. Will keep this in mind for the future.

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py]: Done!

File: cms/jinja2/templates/pages/information_page.html
Comment (sanjeevz3009) [reply cms/jinja2/templates/pages/information_page.html]: Done!

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:12]: Neha and I digged into this. Got it working but have further questions/ doubts. Problem for Monday =)

Comment (sanjeevz3009) [reply cms/standard_pages/models.py:12]: Done!

File: cms/standard_pages/models.py
Comment (zerolab) [inline cms/standard_pages/models.py:34]: suggestion: move this at the top of the class, just after `template`. This helps with a quick scan

Comment (nehakerung) [reply cms/standard_pages/models.py]: done 🫡

**Review by zerolab — APPROVED** [review]
LGTM.

File: cms/static_src/sass/main.scss
Comment (MebinAbraham) [inline cms/static_src/sass/main.scss:19]: I suggest leaving this as is and using the DS component. https://service-manual.ons.gov.uk/design-system/components/video#example.example-video.nunjucks

I suggest raising a separate PR for it.

File: cms/release_calendar/models.py
Comment (MebinAbraham) [inline cms/release_calendar/models.py:53]: How come these changed?. Is something wrong with our formatter setup? There shouldn't be a change between PRs unless some config has changed.

**Review by MebinAbraham — APPROVED** [review]
Nice work both 👍

File: cms/static_src/sass/main.scss
Comment (zerolab) [reply cms/static_src/sass/main.scss:19]: I've added https://jira.ons.gov.uk/browse/CMS-204 for it

File: cms/static_src/sass/main.scss
Comment (zerolab) [inline cms/static_src/sass/main.scss:16]:
```suggestion
    left: 0;  // styleling-disable-line property-disallowed-list
```

should sort this

Comment (sanjeevz3009) [reply cms/static_src/sass/main.scss:16]: Yes, I found this fix yesterday and it fixes the front-end linting issue. I was going to ask about the best practice moving forward, so I didn't push this up.

But now I see that we are using the rules from `stylelint-config-wagtail` and it has that `property-disallowed-list-rule`.

Thanks. Will push up the changes.

Comment (sanjeevz3009) [reply cms/static_src/sass/main.scss:17]: Done!

File: cms/release_calendar/models.py
Comment (sanjeevz3009) [reply cms/release_calendar/models.py:55]: Did some digging. I think it was my setup locally. As I have multiple formatters on VS Code. Black and Ruff.

My default formatter was set up as `Black` and it formatted this file for the line too long issues. I have made the default formatter to `ruff` now.

Now this file is formatted so it's consistent with the `cms/release_calendar/models.py` file.

**Review by zerolab — APPROVED** [review]

File: cms/standard_pages/models.py
Comment (MebinAbraham) [inline cms/standard_pages/models.py:26]: Core storyblock contains equations and embeds, which requires some JS to be loaded which currently are not. See Dan's PR for details: https://github.com/ONSdigital/dis-wagtail/pull/20/files#diff-7c773641945aa15a2cb8c88a31f57b28667ff092ceaefc551f02b50929a3763fR142-R150

**Review by MebinAbraham — CHANGES_REQUESTED** [review]

File: cms/standard_pages/models.py
Comment (zerolab) [reply cms/standard_pages/models.py:26]: 49f9383 may be of help

Comment (sanjeevz3009) [reply cms/standard_pages/models.py:26]: I looked at your PR Dan specifically around the equations and ons_embed. The latest code written in `base.py`, `has_equations ` and `has_ons_embed . I included this locally for this task and also the partial and it all works for me =)

Neha and I tested it.

But I think I'll wait till your PR gets approved and merged first and then I'll pull your code in and use it.

I was reviewing your analysis PR with Neha, and I got some questions, which I'll ask tomorrow on a call =)

Comment (zerolab) [reply cms/standard_pages/models.py:26]: @sanjeevz3009 @nehakerung #20 is now merged, so this is ready to progress

Comment (nehakerung) [reply cms/standard_pages/models.py:26]: Merged and added 🎉

File: cms/jinja2/templates/pages/information_page.html
Comment (zerolab) [inline cms/jinja2/templates/pages/information_page.html:34]:
```suggestion
```

These should not be needed as we include then in `base_page.html` and your page model inherits from `BasePage` which adds the generic `has_equations` / `has_ons_embed`

File: media/documents/file.txt
Comment (zerolab) [inline media/documents/file.txt:1]: rogue file.

could you please add `/media` to `.gitignore` ?

**Review by zerolab — CHANGES_REQUESTED** [review]
A couple of small changes needed (but lovely first feature 💪🏼 )

**Review by MebinAbraham — APPROVED** [review]
Nothing else that's not already been raised.

👍

File: cms/jinja2/templates/pages/information_page.html
Comment (sanjeevz3009) [inline cms/jinja2/templates/pages/information_page.html:29]: Duplicated code here. Now that the partial we need is in the `cms/jinja2/templates/base_page.html` file. Since this page is inheriting from `base_page.html` we don't need this code here.

FYI: In the case that the partial wasn't in the `base_page.html`, you would be importing the partial from `/cms/jinja2/templates/partials/content_scripts.html` and using it here, not rewriting the whole code manually here as that's why we have a partial.

Comment (sanjeevz3009) [reply cms/jinja2/templates/pages/information_page.html:29]: @zerolab Beat me to it!

File: cms/standard_pages/models.py
Comment (sanjeevz3009) [reply cms/standard_pages/models.py:26]: Good job Neha! Added some comments below similar to Dan!

File: cms/jinja2/templates/pages/information_page.html
Comment (nehakerung) [reply cms/jinja2/templates/pages/information_page.html:34]: Understood 🫡

File: cms/jinja2/templates/pages/information_page.html
Comment (nehakerung) [reply cms/jinja2/templates/pages/information_page.html:29]: 🫡 Done

File: media/documents/file.txt
Comment (nehakerung) [reply media/documents/file.txt]: added.

Comment (nehakerung) [reply media/documents/file.txt]: Should've clarified that i just removed the content but have now added /media in .gitignore per @zerolab request.

**Review by zerolab — APPROVED** [review]

File: cms/standard_pages/models.py
Comment (zerolab) [reply cms/standard_pages/models.py:26]: I think this can be resolved now

**Review by MebinAbraham — APPROVED** [review]

File: cms/jinja2/templates/pages/information_page.html
Comment (MebinAbraham) [inline cms/jinja2/templates/pages/information_page.html:13]: To make use of this you need to add this to the content panel:

`InlinePanel("page_related_pages", label="Related pages")`

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**
